Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ActorData.xml

```diff
@@ -41782,4 +41792,136 @@
+        <On Terms="Behavior.AP_YggAutomatedExtractorEssence.Off" Send="Destroy"/>
+        <On Terms="Behavior.AP_YggSpawningPoolEssence.On" Send="Create"/>
+        <On Terms="Behavior.AP_YggSpawningPoolEssence.Off" Send="Destroy"/>
+        <On Terms="Behavior.AP_YggRoachWarrenEssence.On" Send="Create"/>
+        <On Terms="Behavior.AP_YggRoachWarrenEssence.Off" Send="Destroy"/>
+        <On Terms="Behavior.AP_YggBanelingNestEssence.On" Send="Create"/>
+        <On Terms="Behavior.AP_YggBanelingNestEssence.Off" Send="Destroy"/>
+        <On Terms="Behavior.AP_YggEvolutionChamberEssence.On" Send="Create"/>
+        <On Terms="Behavior.AP_YggEvolutionChamberEssence.Off" Send="Destroy"/>
+        <On Terms="Behavior.AP_YggSpineCrawlerEssence.On" Send="Create"/>
+        <On Terms="Behavior.AP_YggSpineCrawlerEssence.Off" Send="Destroy"/>
+        <On Terms="Behavior.AP_YggSporeCrawlerEssence.On" Send="Create"/>
+        <On Terms="Behavior.AP_YggSporeCrawlerEssence.Off" Send="Destroy"/>
+        <On Terms="Behavior.AP_YggInfestationPitEssence.On" Send="Create"/>
+        <On Terms="Behavior.AP_YggInfestationPitEssence.Off" Send="Destroy"/>
+        <On Terms="Behavior.AP_YggNydusNetworkEssence.On" Send="Create"/>
+        <On Terms="Behavior.AP_YggNydusNetworkEssence.Off" Send="Destroy"/>
+        <On Terms="Behavior.AP_YggUltraliskCavernEssence.On" Send="Create"/>
+        <On Terms="Behavior.AP_YggUltraliskCavernEssence.Off" Send="Destroy"/>
+        <On Terms="Behavior.AP_YggPredatorNestEssence.On" Send="Create"/>
+        <On Terms="Behavior.AP_YggPredatorNestEssence.Off" Send="Destroy"/>
+        <On Terms="Behavior.AP_YggHydraliskDenEssence.On" Send="Create"/>
+        <On Terms="Behavior.AP_YggHydraliskDenEssence.Off" Send="Destroy"/>
+        <On Terms="Behavior.AP_YggGreaterHydraliskDenEssence.On" Send="Create"/>
+        <On Terms="Behavior.AP_YggGreaterHydraliskDenEssence.Off" Send="Destroy"/>
+        <On Terms="Behavior.AP_YggSpireEssence.On" Send="Create"/>
+        <On Terms="Behavior.AP_YggSpireEssence.Off" Send="Destroy"/>
+        <On Terms="Behavior.AP_YggGreaterSpireEssence.On" Send="Create"/>
+        <On Terms="Behavior.AP_YggGreaterSpireEssence.Off" Send="Destroy"/>
+        <HostSiteOps Ops="SOpAttachOrigin"/>
+        <Model value="ConsumptionImpact"/>
+    </CActorModel>
+    <CActorUnit id="AP_Yggdrasil" parent="GenericUnitMorph" unitName="AP_Overlord">
+        <Aliases value="_UnitLarge"/>
+        <Macros value="PhysicsDeathsSwarmAir"/>
+        <Macros value="PhysicsDeathsVoidAir"/>
+        <On index="0" Terms="UnitBirth.AP_Yggdrasil"/>
+        <On index="1" Terms="UnitBirth.AP_Yggdrasil"/>
+        <On index="2" Terms="UnitRevive.AP_Yggdrasil"/>
+        <On index="3" Terms="UnitRevive.AP_Yggdrasil"/>
+        <On index="4" Terms="UnitConstruction.AP_Yggdrasil.Start"/>
+        <On index="5" Terms="UnitConstruction.AP_Yggdrasil.Finish"/>

```

Comment:
If you set `unitName` to `AP_Yggdrasil`, you should be able to simplify a lot

Reply:
Cleaned up some of the junk, specifically "On index = "x" " lines.